tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -43,17 +42,12 @@ async def dummy_task(*_: Any) -> None:
     return task
 
 
-async def dummy_receive(receiving_tasks: List[asyncio.Future]) -> None:
-    """A dummy receive method that simply waits for the first task to complete and returns the done and pending."""
-    await asyncio.wait(receiving_tasks, return_when=asyncio.FIRST_COMPLETED)
-
-
 class TestLedgerConnection:
     """Test `LedgerConnection` class."""
 
     @pytest.mark.asyncio
-    async def test_not_hanging(self) -> None:
-        """Test that the connection does not hang and that the tasks cannot get blocked."""
+    async def test_hanging(self) -> None:
```

Comment:
this test is use to showcase the problem

Reply:
Was anticipating the fix :)